src/vivarium_gates_bep/tools/make_bw_risk_correlation.py

```diff
@@ -0,0 +1,65 @@
+"""Application functions for producing specification files from which we derive birth weight risk correlation."""
+import numpy as np
+import pandas as pd
+
+from pathlib import Path
+
+from vivarium import InteractiveContext
+from vivarium_gates_bep.tools.make_specs import build_model_specifications
+
+
+LARGER_THAN_LARGEST_BABY_ON_RECORD = 25 * 454
+
+
+def empirical_percentile(x, samples):
+    return np.mean(x >= samples)
+
+
+def create_bw_rc_data(spec_file: str):
+    sim = InteractiveContext(spec_file)
+    df = pd.DataFrame()
+    df['birth_weights'] = sim.get_population().birth_weight
+
+    # rank birth weights
```

Comment:
Couldn't you sort the df by birth weight and then just increment to get the percentile rather than applying this function which is O(n^2)

Reply:
Yup, cleaner and less work. Thanks.